src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -57,10 +57,7 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
                                 // PROTOTYPE: this should check for init-only, but that isn't a separate feature yet
                                 if (!(sym is SynthesizedRecordPropertySymbol))
                                 {
-                                    hasErrors = true;
-                                    diagnostics.Add(
-                                        ErrorCode.ERR_WithMemberIsNotRecordProperty,
-                                        location);
+                                    goto default;
```

Comment:
This is actually a behavior change: before we'd error and set `member = sym`, now we don't set `member = sym`. That's why I was suggesting doing it after that part. If there's no test that demonstrates this, consider adding one.

Reply:
Oops, good catch. I'll verify